// token-auth-check/tokenCheck.js
//Checks to see if the token given by the user is legitimate
function tokenCheck(){
	this.pool = require('../database/DBPool');
	this.graph = require('fbgraph');
	this.parser = require('json-parser');
}

//Checks the user token
tokenCheck.prototype.checkToken = function(token, cb){
	var query = "select * from users where uid=" + token;
	var currentRef = this;
	this.pool.sendQuery(query, function(response, fields){ //Queries db to see if token is there
		if(response.length < 1 ){
			console.log("User not found");
			cb("Invalid user");
		}
		else{
			currentRef.fbCheck(response[0].fbtoken, cb); //Next checks to see if Facebook token is valid
		}
	});
}

//Checks to see if the facebook token is valid
tokenCheck.prototype.fbCheck = function(fbtoken, cb){
	var facebookParameters = "fields=first_name,last_name,birthday,gender&"; //Parameters for GET request
	this.graph
	  	.setAccessToken(fbtoken) //Setting token to one retrieved from db
	  	.get("/me?" + facebookParameters, function(err, data) { //Sending graph API request
	  	console.log(data);
		if(err){
	  		console.log("Token is invalid");
	  		cb(data);
	  	}
	  	else{
			console.log("Token is valid, continuing on");
			cb("Valid Everything");
	  	}
		});
}

module.exports = tokenCheck;
