// token-auth-check/tokenCheck.js
//Checks to see if the token given by the user is legitimate
function tokenCheck(){
	this.pool = require('../database/DBPool');
	this.graph = require('fbgraph');
	this.parser = require('json-parser');
}

tokenCheck.prototype.checkToken = function(token, cb){
	var query = "select * from users where uid=" + token;
	var currentRef = this;
	this.pool.sendQuery(query, function(response, fields){
		//We don't care about the fields at all, just focus on the response
		// console.log(response[0].fbtoken);
		if(response.length < 1 ){
			console.log("User not found");
			cb("Invalid user");
		}
		else{
			//Now we need to check the fbtoken
			currentRef.fbCheck(response[0].fbtoken, cb);
		}
	});
}

//Gets the legitimate fbtoken from the database based on the token entered
//Runs a fb graph api request to see if it's valid or not
tokenCheck.prototype.fbCheck = function(fbtoken, cb){
	//Making the graph call

	var facebookParameters = "fields=first_name,last_name,birthday,gender&";
	this.graph
	  	.setAccessToken(fbtoken)
	  	.get("/me?" + facebookParameters, function(err, data) {
	  	console.log(data);
		if(err){
	  		console.log("Token is invalid");
	  		cb(data);
	  	}
	  	else{
			console.log("Token is valid, continuing on");
			cb("Valid Everything");
	  	}
		});
}

module.exports = tokenCheck;
