// database/DBConnection.js

function DBConnection() {	
	this.mysql = require("mysql");
	this.dbinfo = require('./connection.json');
	this.con = this.mysql.createConnection(this.dbinfo);
	
	this.con.connect(function(err){
  		if(err){
    		console.log('Error connecting to Db');
    		return;
  		}
  		console.log('Connection established');
	});
}

//Send an arbitary query to the database
DBConnection.prototype.sendQuery = function(queryString, callBack, httpResponse){
	console.log("The query you entered is " + queryString); //DEBUG
	var query = this.con.query(queryString, function(err, response){
		if(err){
			callBack(err, httpResponse);
		}
		else{
			callBack(response, httpResponse);
		}
	});
	this.endConnection();

}

//End the connection to the database
DBConnection.prototype.endConnection = function(){
	//This is just a function that ends the connection to the server
	this.con.end(function(err){
		if(err){
			console.log(err);
		}
		//console.log('The connection to the server has been terminated');
	});
}

module.exports = DBConnection;
