// database/DBConnection.js
//Class when instantiated connects to database
function DBConnection() {	
	this.mysql = require("mysql"); //Import mysql module
	this.dbinfo = require('./connection.json'); //Authentication information for database
	this.pool = this.mysql.createPool(this.dbinfo); //Create db connection pool 
}

//Function to execute query in database
DBConnection.prototype.sendQuery = function(queryString, callBack, httpResponse) {
	//Obtain a free connection to the database
	var globPool = this.pool;
	this.pool.getConnection(function(err, connection) {
		//Submit query to database
		var query = globPool.query(queryString, function(err, rows) {		
			//Check query results for an error
			if(err){
				//Log error to console
				console.log(err);

				//Send error response to callback function
				callBack(err, httpResponse);
			}
			else{
				//Send query results back up through callback function
				callBack(rows, httpResponse);
			}

			//Free db connection to service queued queries
		 	connection.release();
		});
	});
}

//Close all connections to database
DBConnection.prototype.endConnection = function(){
	this.pool.end(function (err) {
		//Connections are closed
		//Check for error 
		if(err) {
			console.log(err);	
		}
	});
}

//Return reference to connection
module.exports = DBConnection;
