// database/DBConnection.js
//Class when instantiated connects to database
function DBConnection() {	
	this.mysql = require("mysql"); //Import mysql module
	this.dbinfo = require('./connection.json'); //Authentication information for database
	this.con = this.mysql.createConnection(this.dbinfo); //Create db connection 

	//Check connection for errors
	this.con.connect(function(err){ 
  		//Prompt for error or successful db connection
		if(err){
    		console.log('Error connecting to Db');
    		return;
  		}
  		else
  			console.log('Connection established'); //Success
	});
}

//Function to execute query in database
DBConnection.prototype.sendQuery = function(queryString, callBack, httpResponse) {
	//Submit query to database
	var query = this.con.query(queryString, function(err, response) {
		//Check query results for an error
		if(err){
			//Log error to console
			console.log(err);

			//Send error response to callback function
			callBack(err, httpResponse);
		}
		else{
			//Send query results back up through callback function
			callBack(response, httpResponse);
		}
	});
}

//Close connection to database
DBConnection.prototype.endConnection = function(){
	//Execute call to kill connection 
	this.con.end(function(err){
		//Check for error 
		if(err){
			//Display error
			console.log(err);
		}
	});
}

//Return reference to connection
module.exports = DBConnection;
