// database/DBPool.js
//Instantiates database connection
function DBPool() {	
	this.mysql = require("mysql"); 
	this.dbinfo = require('./connection.json'); //Authentication information for database
	this.pool = this.mysql.createPool(this.dbinfo); //Create db connection pool 
}

//Function to execute query in database
DBPool.prototype.sendQuery = function(queryString, cb, httpResponse) {
	//Obtain a free connection to the database
	var globPool = this.pool;
	this.pool.getConnection(function(err, connection) {
		//Submit query to database
		var query = globPool.query(queryString, function(err, rows) {		
			//Check query results for an error
			if(err){
				//Log error to console
				console.log(err);

				//Send error response to cb function
				cb(err, httpResponse);
			}
			else{
				//Send query results back up through cb function
				cb(rows, httpResponse);
			}

			//Free db connection to service queued queries
		 	connection.release();
		});
	});
}

//Close all connections to database
DBPool.prototype.endConnection = function(){
	this.pool.end(function (err) {
		//Connections are closed
		//Check for errors
		if(err) {
			console.log(err);
		}
	});
}

//Return reference to connection
module.exports = DBPool;
